ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/TraceReplayBlockTransactions.java

```diff
@@ -104,122 +97,86 @@ protected Object resultByBlockNumber(
         .orElse(null);
   }
 
-  private Object traceBlock(final Block block, final TraceTypeParameter traceTypeParameter) {
+  private ArrayNode traceBlock(final Block block, final TraceTypeParameter traceTypeParameter) {
 
-    if (block == null || block.getBody().getTransactions().isEmpty()) {
+    if (block == null) {
       return emptyResult();
     }
 
     final Set<TraceTypeParameter.TraceType> traceTypes = traceTypeParameter.getTraceTypes();
-    final TraceOptions traceOptions =
-        new TraceOptions(false, false, traceTypes.contains(VM_TRACE) || traceTypes.contains(TRACE));
 
+    final BlockHeader header = block.getHeader();
     return Tracer.processTracing(
-            blockchainQueriesSupplier.get(),
-            Optional.of(block.getHeader()),
-            mutableWorldState -> {
-              return blockTracerSupplier
-                  .get()
-                  .trace(mutableWorldState, block, new DebugOperationTracer(traceOptions))
-                  .map(BlockTrace::getTransactionTraces)
-                  .map((traces) -> generateTracesFromTransactionTrace(traces, block, traceTypes));
+            getBlockchainQueries(),
+            Optional.of(header),
+            traceableState -> {
+              ArrayNodeWrapper resultArrayNode = new ArrayNodeWrapper(MAPPER.createArrayNode());
+
+              final ProtocolSpec protocolSpec = protocolSchedule.getByBlockHeader(header);
+              final MainnetTransactionProcessor transactionProcessor =
+                  protocolSpec.getTransactionProcessor();
+              final ChainUpdater chainUpdater = new ChainUpdater(traceableState);
+
+              final TransactionSource transactionSource = new TransactionSource(block);
+              final LabelledMetric<Counter> outputCounter =
+                  new PrometheusMetricsSystem(BesuMetricCategory.DEFAULT_METRIC_CATEGORIES, false)
+                      .createLabelledCounter(
+                          BesuMetricCategory.BLOCKCHAIN,
+                          "transactions_traceblock_pipeline_processed_total",
```

Comment:
should the name for this metric be different from the one in TraceBlock

Reply:
I changed the name, thanks.